foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/security/JCEEncryptor.java

```diff
@@ -125,10 +127,15 @@ private static SecretKey getAESGCMMultitasker() throws Exception {
         private static byte[] getIvGCM() {
             byte[] ivGCM = new byte[IV_GCM_LENGTH];
             SecureRandom random = null;
-            try {
-                random = SecureRandom.getInstanceStrong();
-            } catch (NoSuchAlgorithmException e) {
-                throw new RuntimeException(e);
+            String useStrongRNG = PrivilegedAccessHelper.getSystemProperty(SystemProperties.SECURITY_ENCRYPTOR_USE_STRONG_RANDOM_NUMBER_GENERATOR);
+            if (Boolean.parseBoolean(useStrongRNG)) {
```

Comment:
I'd like see there validation for input see e.g.
https://github.com/eclipse-ee4j/eclipselink/blob/master/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/asm/ASMFactory.java#L268
and throw `throw ValidationException.incorrect.....` to point user what is wrong.

Reply:
Thank you for your advice.
I tried to find examples of boolean value validation to meet your expectation, but I could not find a good example...

- https://github.com/eclipse-ee4j/eclipselink/blob/5.0.0-B04/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConversionManager.java#L115
- https://github.com/eclipse-ee4j/eclipselink/blob/5.0.0-B04/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/Helper.java#L1133
- https://github.com/eclipse-ee4j/eclipselink/blob/5.0.0-B04/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/mappings/UnidirectionalOneToManyMapping.java#L401
- https://github.com/eclipse-ee4j/eclipselink/blob/5.0.0-B04/foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConcurrencyUtil.java#L68-L73

For example, `java.lang.Boolean.parseBoolean()` can be used to parse boolean values, but the method cannot be used for validation since it does not throw exceptions.
Validation could be achieved as in `if (!"true".equalsIgnoreCase(value) && !"false".equalsIgnoreCase(value)) { throw ... }`, but such code is not pretty.
Could you suggest a good way to validate boolean values if you have?

Reply:
Sorry I don't have anything better. Similar pattern `"true".equals....` is used sometimes inside EclipseLink e.g.
https://github.com/eclipse-ee4j/eclipselink/blob/master/jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/EntityManagerImpl.java#L275 
Just use this "not pretty" pattern and throw `org.eclipse.persistence.exceptions.ValidationException#invalidBooleanValueForProperty`.

Reply:
I have reflected your comment now. Please check it.
If an invalid value is specified, an exception like below is thrown.
```
Exception [EclipseLink-7278] (Eclipse Persistence Services - 5.0.0.v202412160411-a5af5837377bcad2adc674a0682717d24a436f69): org.eclipse.persistence.exceptions.ValidationException
Exception Description: The specified boolean value [aaa] for the persistence property [eclipselink.security.encryptor.use.strong.random.number.generator] is invalid, the value must either be "true" or "false".
```